src/Revision.php

```diff
@@ -0,0 +1,135 @@
+ *
+ * Revision is the sole interface for manipulating a session's attribute table.
+ *
+ * The underlying session may be "local", i.e. owned by a peer running in this
+ * process, or "remote", owned by a different peer.
+ *
+ * For remote sessions, operations may require network IO. Deadlines are
+ * honoured for all methods that accept a context.
+ */
+interface Revision
+{
+    /**
+     * The session reference, which holds the session ID and the revision number
+     * represented by this instance.
+     *
+     * @return Reference The session reference.
+     */
+    public function reference(): Reference;
+
+    /**
+     * Refresh returns the latest revision of the session.
+     *
+     * @return Revision Latest revision of the session.
+     *
+     * @throws NotFoundException If the session has been closed and revision is invalid.
+     */
+    public function refresh(Context $context): Revision;
+
+    /**
+     * Get returns the attribute with key $key from the attribute table.
+     *
+     * The returned attribute is guaranteed to be correct as of
+     * Reference::revision().
+     *
+     * Non-existent attributes are equivalent to empty attributes, therefore it
+     * is not an error to request a key that has never been created.
+     *
+     * Peers do not always have a copy of the complete attribute table. If the
+     * attribute value is unknown it is fetched from the owning peer.
+     *
+     * To fetch the attribute value at the later revision, first call refresh()
+     * then retry the get() on the newer revision.
+     *
+     * @param Context $context
+     * @param string  $key
+     *
+     * @return Attribute The attribute matching key $key.
+     *
+     * @throws StaleFetchException If the attribute can not be retreived because it has already been modified.
+     * @throws NotFoundException   If the session has been closed and the revision can not be queried.
+     */
+    public function get(Context $context, string $key): Attribute;
+
+    /**
+     * GetMany returns the attributes with keys in $keys from the attribute
+     * table.
+     *
+     * The returned attributes are guaranteed to be correct as of
+     * Reference::revision().
+     *
+     * Non-existent attributes are equivalent to empty attributes, therefore it
+     * is not an error to request keys that have never been created.
+     *
+     * Peers do not always have a copy of the complete attribute table. If any
+     * of the attribute values are unknown they are fetched from the owning peer.
+     *
+     * If any of the attributes can not be retreived because they have already
+     * been modified, StaleFetchException will be thrown. To fetch the attribute
+     * values at the later revision, first call refresh() then retry the
+     * getMany() on the newer revision.
+     *
+     * @return AttributeTable With the the keys $keys.
+     *
+     * @throws StaleFetchException If the attributes were not the latest revision.
```

Comment:
The description of `StaleFetchException` from `get()` is more accurate:

`@throws StaleFetchException If the attributes can not be retrieved because one or more of them has already been modified.`

I've also misspelled "retrieve" all over the place in the Go stuff, by the looks of it. >_<